src/library_pthread.js

```diff
@@ -780,12 +759,13 @@ var LibraryPThread = {
       // stack size if not specified is 2 MB, so follow that convention.
       stackSize = {{{ DEFAULT_PTHREAD_STACK_SIZE }}};
     }
-    // If allocatedOwnStack == true, then the pthread impl maintains the stack allocation.
-    var allocatedOwnStack = stackBase == 0;
-    if (allocatedOwnStack) {
+    // If stackBase is zero then we allocate a new stack region and mark it as
+    // owned.
+    if (stackBase == 0) {
       // Allocate a stack if the user doesn't want to place the stack in a
       // custom memory area.
       stackBase = _memalign({{{ STACK_ALIGN }}}, stackSize);
+      Atomics.store(HEAPU32, (pthread_ptr + {{{ C_STRUCTS.pthread.stack_owned }}}) >> 2, 1);
```

Comment:
Do we have a guarantee that this is initialized to 0 (so we don't need to set it in the other case)?

Reply:
Yes the entire pthread struct is zero initialized in `__pthread_create` just after its allocated.